tests/test_flats.py

```diff
@@ -0,0 +1,50 @@
+"""
+Unit test code for flat production code.
+"""
+from collections import namedtuple
+import unittest
+import numpy as np
+import galsim
+import desc.imsim
+
+TreeRingInfo = namedtuple('TreeRingInfo', ['center', 'func'])
+
+class GsDetector:
+    """
+    Minimal implementation of an interface-compatible version
+    of GalSimDetector for testing the flat production code.
+    """
+    def __init__(self):
+        self.xMaxPix = 256
+        self.yMaxPix = 256
+        self.xMinPix = 1
+        self.yMinPix = 1
+        self.wcs = galsim.wcs.PixelScale(0.2)
+        self.tree_rings \
+            = TreeRingInfo(galsim.PositionD(0, 0),
+                           galsim.SiliconSensor.simple_treerings(0.26, 47))
```

Comment:
Why test with the simple treerings as opposed to the model being used in the code?

Reply:
It's generally a good idea to keep these things as decoupled as possible.  If someone does some work on the treering code in this package, any bugs in that new work shouldn't prevent the flat production code from passing its tests.